### PR TITLE
doc/md: clickhouse guide

### PR DESCRIPTION
**Pull request by hilakashai**

### CODE REVIEW

**Comment by giautm** [comment]
@hilakashai: There is a broken link.
```
Note: it's possible to ignore broken links with the 'onBrokenLinks' Docusaurus configuration, and let the build pass.

Exhaustive list of all broken links found:

- On source page path = /guides/clickhouse/getting-started:
   -> linking to /docs
```

File: doc/md/guides/clickhouse/getting-started.mdx [hunk 1]
```diff
@@ -0,0 +1,533 @@
+---
+id: getting-started-clickhouse
+title: Manage ClickHouse with Atlas
+slug: /guides/clickhouse/getting-started
```
Comment (rotemtam) [inline doc/md/guides/clickhouse/getting-started.mdx:4]: I think /guides/clickhouse

Is enough

File: doc/md/guides/clickhouse/getting-started.mdx [hunk 2]
```diff
@@ -0,0 +1,533 @@
...
+```sql
+CREATE TABLE `users` (
+  `id` UInt64,
+  `name` String NOT NULL,
+  PRIMARY KEY (`id`)
+) ENGINE = MergeTree() ORDER BY id;
+```
+
+The `atlas schema inspect` command supports reading the database description provided by a [URL](/concepts/url) and outputting it in
+three different formats: [Atlas DDL](/atlas-schema/hcl.mdx) (default), SQL, and JSON. In this guide, we will
```
Comment (rotemtam) [inline doc/md/guides/clickhouse/getting-started.mdx:45]: It supports more than 3

File: doc/md/guides/clickhouse/getting-started.mdx [hunk 3]
```diff
@@ -0,0 +1,533 @@
...
+to the inspect command:
+
+```shell
+atlas schema inspect -u "clickhouse://localhost:9000/example" -w
+```
+
+![clickhouse-inspect](https://atlasgo.io/uploads/clickhouse/images/clickhouse-inspect.png)
+
+## Declarative Migrations
+The declarative approach requires the user to define the desired end schema, and Atlas provides a safe way to
```
Comment (rotemtam) [inline doc/md/guides/clickhouse/getting-started.mdx:126]: The approach allows or enables  - require sounds limiting

File: doc/md/guides/clickhouse/getting-started.mdx [hunk 4]
```diff
@@ -0,0 +1,533 @@
...
+
+Now that our _desired state_ has changed, to apply these changes to our database, Atlas will plan a migration for us
+by running the `atlas schema apply` command:
+
+<Tabs>
+<TabItem value="hcl" label="Atlas DDL (HCL)" default>
+
+```shell
+atlas schema apply \
+-u "clickhouse://localhost:9000/demo" \
```
Comment (rotemtam) [inline doc/md/guides/clickhouse/getting-started.mdx:212]: Use a dev db

File: doc/md/guides/clickhouse/getting-started.mdx [hunk 5]
```diff
@@ -0,0 +1,533 @@
...
+required migrations.
+
+In this guide, we will dive into setting up Atlas for ClickHouse, and introduce the different workflows available.
+
+## Prerequisites
+1. A ClickHouse database
+2. Docker
+3. Atlas installed on your machine:
+<InstallationInstructions />
+
```
Comment (rotemtam) [inline doc/md/guides/clickhouse/getting-started.mdx:31]: Users must login to use CH

File: doc/md/guides/clickhouse/getting-started.mdx [hunk 6]
```diff
@@ -0,0 +1,533 @@
...
+Now that we have our first migration, we can apply it to a database. There are multiple ways to accomplish this, with
+most methods covered in the [guides](/guides) section. In this example, we'll demonstrate how to push migrations to
+[Atlas Cloud](https://atlasgo.cloud), much like how Docker images are pushed to Docker Hub.
+
+<div style={{textAlign: 'center'}}>
+    <img src="https://atlasgo.io/uploads/clickhouse/images/first-push.png" alt="clickhouse migrate push" width="100%"/>
+    <p style={{fontSize: 12}}>Migration Directory created with <code>atlas migrate push</code></p>
+</div>
+
+Firstly, you'll need to [log in to Atlas](https://auth.atlasgo.cloud/signup). If it's your first time, you'll be
```
Comment (rotemtam) [inline doc/md/guides/clickhouse/getting-started.mdx:309]: Already logged in to use ch

**Review by rotemtam — COMMENTED** [review]
Well done, see comments

File: doc/md/guides/clickhouse/getting-started.mdx [hunk 7]
```diff
@@ -0,0 +1,536 @@
+---
```
Comment (rotemtam) [inline doc/md/guides/clickhouse/getting-started.mdx:1]: also move the file to /guides/clickhouse.mdx

File: doc/md/guides/clickhouse/getting-started.mdx [hunk 8]
```diff
@@ -0,0 +1,536 @@
...
+
+#### Enter: Atlas
+Atlas helps developers manage their database schema as code - abstracting away the intricacies of database schema
+management. With Atlas, users provide the desired state of the database schema and Atlas automatically plans the
+required migrations.
+
+In this guide, we will dive into setting up Atlas for ClickHouse, and introduce the different workflows available.
+
+## Prerequisites
+1. A ClickHouse database
```
Comment (rotemtam) [inline doc/md/guides/clickhouse/getting-started.mdx:27]:
```suggestion

1. A ClickHouse database
```

File: doc/md/guides/clickhouse/getting-started.mdx [hunk 9]
```diff
@@ -0,0 +1,536 @@
...
+```
+
+</TabItem>
+</Tabs>
+
+## Inspecting our Database
+
+Let's start off with a minimal database with a `users` table and an `id` as the primary key.
+
+```sql
```
Comment (rotemtam) [inline doc/md/guides/clickhouse/getting-started.mdx:65]: How is the initial database created? 

How is this schema created in clickhouse?

File: doc/md/guides/clickhouse/getting-started.mdx [hunk 10]
```diff
@@ -0,0 +1,536 @@
...
+  primary_key {
+    columns = [column.id]
+  }
+}
+schema "demo" {
+  engine  = sql("Memory")
+}
+```
+
+This block represents a [table](/atlas-schema/hcl.mdx#table) resource with `id`, and `name`
```
Comment (rotemtam) [inline doc/md/guides/clickhouse/getting-started.mdx:110]: There are two blocks in this snippet

File: doc/md/guides/clickhouse/getting-started.mdx [hunk 11]
```diff
@@ -0,0 +1,536 @@
...
+columns. The `schema` field references the `demo` schema that is defined elsewhere
```
Comment (rotemtam) [inline doc/md/guides/clickhouse/getting-started.mdx:111]: elsewhere is right below, sounds weird

File: doc/md/guides/clickhouse/getting-started.mdx [hunk 12]
```diff
@@ -0,0 +1,536 @@
...
+to the inspect command:
+
+```shell
+atlas schema inspect -u "clickhouse://localhost:9000/example" -w
+```
+
+![clickhouse-inspect](https://atlasgo.io/uploads/clickhouse/images/clickhouse-inspect.png)
+
+## Declarative Migrations
+The declarative approach allows the user to define the desired end schema, and Atlas provides a safe way to
```
Comment (rotemtam) [inline doc/md/guides/clickhouse/getting-started.mdx:155]:
```suggestion
## Declarative Migrations

The declarative approach allows the user to define the desired end schema, and Atlas provides a safe way to
```

Comment (rotemtam) [reply doc/md/guides/clickhouse/getting-started.mdx:155]: The declarative approach allows users to manage their database schemas by simply..

File: doc/md/guides/clickhouse/getting-started.mdx [hunk 13]
```diff
@@ -0,0 +1,536 @@
...
+For a more detailed description of the `atlas schema apply` command refer to our documentation
+[here](/declarative/apply).
+:::
+
+To ensure that the changes have been made to the schema, let's run the `inspect` command with the `-w` flag once more
+and view the ERD:
+
+![atlas-schema](https://atlasgo.io/uploads/clickhouse/images/clickhouse-two-tables.png)
+
+## Versioned Migrations
```
Comment (rotemtam) [inline doc/md/guides/clickhouse/getting-started.mdx:270]: space between all of the rest of the subheadings please

File: doc/md/guides/clickhouse/getting-started.mdx [hunk 14]
```diff
@@ -0,0 +1,536 @@
...
+<TabItem value="atlas.sum" label="atlas.sum">
+
+```shell
+h1:NrYfakYn4W3xiS9VqcP98sqvgLUPn9pJpxFndh1GWsQ=
+20240130122951.sql h1:R+eBw2nOk2+VLBvJ/dn49OphxzfxoWEBUxAy4Zp3VAE=
+```
+</TabItem>
+</Tabs>
+
+The migration file represents the current state of our database, and the sum file is in charge of maintaining the integrity
```
Comment (rotemtam) [inline doc/md/guides/clickhouse/getting-started.mdx:326]: the file isn't in charge :-) it is used by atlas to..

File: doc/md/guides/clickhouse/getting-started.mdx [hunk 15]
```diff
@@ -0,0 +1,536 @@
...
+    <img src="https://atlasgo.io/uploads/clickhouse/images/second-push.png" alt="clickhouse migrate push" width="100%"/>
+    <p style={{fontSize: 12}}>Migration Directory created with <code>atlas migrate push</code></p>
+</div>
+
+## Next Steps
+
+In this guide we learned about the declarative and versioned workflows, and how to use Atlas to generate migrations,
+push them to an Atlas workspace and apply them to databases.
+
+Next, we recommend you [set up CI](/cloud/setup-ci) for your migration directory, to ensure that all changes applied to
```
Comment (rotemtam) [inline doc/md/guides/clickhouse/getting-started.mdx:531]: Next steps:
* Full docs for clickhouse HCL schema
* How to set up CI
* Deploying changes with Terraform, Kubernetes
* Learn about modern CI/CD for databases

File: doc/md/guides/clickhouse/clickhouse.mdx [hunk 16]
```diff
@@ -0,0 +1,553 @@
...
+  `id` UInt64,
+  `name` String NOT NULL,
+  PRIMARY KEY (`id`)
+) ENGINE = MergeTree() ORDER BY id;
+```
+
+To create the table above on our local database, we can run the following command:
+
+```sql
+ docker exec -it atlas-demo clickhouse-client --query 'CREATE TABLE demo.users (id UInt64, name String NOT NULL, PRIMARY KEY (id)) ENGINE = MergeTree() ORDER BY id'
```
Comment (rotemtam) [inline doc/md/guides/clickhouse/clickhouse.mdx:84]: 💯

**Review by rotemtam — APPROVED** [review]

**Pull request merged**